src/expr/src/relation.rs

```diff
@@ -1107,13 +1107,18 @@ impl MirRelationExpr {
                                         {
                                             if first_id == second_id {
                                                 result.extend(
-                                                    inputs[0].typ().keys.drain(..).filter(|key| {
-                                                        key.iter().all(|c| {
-                                                            outputs.get(*c) == Some(c)
-                                                                && base_projection.get(*c)
-                                                                    == Some(c)
-                                                        })
-                                                    }),
+                                                    input_keys
+                                                        .next()
+                                                        .unwrap()
+                                                        .clone()
```

Comment:
Rather than `.clone()`ing up front, you can put a `.cloned()` at the very end, so we only copy the ones we want to keep.
```suggestion
```